exporter/exporter.go

```diff
@@ -158,74 +160,42 @@ func (e *Exporter) makeRegistry(ctx context.Context, client *mongo.Client, topol
 
 	// If we manually set the collection names we want or auto discovery is set.
 	if (len(e.opts.CollStatsNamespaces) > 0 || e.opts.DiscoveringMode) && e.opts.EnableCollStats && limitsOk && requestOpts.EnableCollStats {
-		cc := collstatsCollector{
-			ctx:             ctx,
-			client:          client,
-			collections:     e.opts.CollStatsNamespaces,
-			compatibleMode:  e.opts.CompatibleMode,
-			discoveringMode: e.opts.DiscoveringMode,
-			logger:          e.opts.Logger,
-			topologyInfo:    topologyInfo,
-		}
-		registry.MustRegister(&cc)
+		cc := newCollectionStatsCollector(ctx, base,
```

Comment:
We are adding collectors like this one (StatsCollector) to the the registry and Prometheus then will call each collector on every scrape. By having one single base instance and passing it to every collector we are making them to share the lock and the metrics cache. 
AFAIK, Prometheus call each collector sequentially but if they change this behavior and call collectors concurrently each collector will lock the others. 
What you think about call the `newXXXCollector` with `client` and any other parameter and create a new and independent base for each collector?

Reply:
+1